mm/pmm.c

```diff
@@ -491,13 +568,13 @@ static void merge_frames(frame_t *first) {
 
```

Comment:
Alternatively, don't we need to take the lock here?

Reply:
All `static` functions are considered internal to the PMM implementation. It is better to take locks as externally as possible (external interfaces, potentially shared by different code other than PMM) to avoid locking/unlocking mishaps. Hence the extra interfaces to handle locking consistently. The `merge_frames()` and `split_frames()` are considered internal and assumed to operate with a taken lock.